src/util-cache.ts

```diff
@@ -0,0 +1,106 @@
+/** A value that will expire after a certain amount of time. */
+export type Transient<T> = readonly [T, number];
```

Comment:
Wouldn't it be clearer if this was a `{value: T, lifetime: number}` instead?

Sure, it's only for internal use, but still.

Reply:
I just added another argument, see: https://github.com/mausworks/tyin/pull/10/commits/c312364f637ed4c8a601cdae91c93ab38032415d

It had multiple benefits.